libs/features/mas/src/price/utilities.js

```diff
@@ -252,15 +309,67 @@ const formatRegularPrice = ({ commitment, term, ...data }) =>
     formatPrice(data, recurrenceTerms[commitment]?.[term]);
 
 /**
- * Formats annual price.
- * @param { import('./types').PriceData } data
+ * Creates a function that calculates the annual price with promotion applied.
+ *
+ * @param {object} data - The data object containing price and priceWithoutDiscount
+ * @returns {function(number): number} A function that takes a monthly price and returns the calculated annual price with promotion
+ *
  */
 const formatAnnualPrice = (data) => {
-    const { commitment, term } = data;
+    const {
+        commitment,
+        instant,
+        price,
+        originalPrice,
+        priceWithoutDiscount,
+        promotion,
+        quantity = 1,
+        term,
+    } = data;
     if (commitment === Commitment.YEAR && term === Term.MONTHLY) {
-        return formatPrice(data, RecurrenceTerm.YEAR, (price) => price * 12);
+        if (!promotion) {
+            return formatPrice(data, RecurrenceTerm.YEAR, getAnnualPrice);
+        }
+        const {
+            displaySummary: { outcomeType, duration, minProductQuantity = 1 } = {},
+        } = promotion;
+        switch (outcomeType) {
+            case 'PERCENTAGE_DISCOUNT': {
+                if (
+                    quantity >= minProductQuantity &&
+                    isPromotionActive(promotion, instant)
```

Comment:
@yesil  What if `outcomeType` is `'PERCENTAGE_DISCOUNT'` but this condition 
`quantity >= minProductQuantity && isPromotionActive(promotion, instant)` in IF is false?
Then I guess `formatAnnualPrice()` will return undefined, it will not go to `default:`, that might create problems since it is expected that this function will return some object but I don't know if that is possible in real world. Just asking.

Reply:
@bozojovicic it goes to `default:` default because there is no `break;`.

these cases are covered by unit tests:

https://github.com/adobecom/milo/pull/3492/files#diff-66faad1f378a7658cebf33b4a9fa66d2ec4ee53039ec0e6fd5ce41930c92c13eR343-R374